### ISSUE TITLE
Anti-DDoS with Cloudflare preventing app use

### ISSUE THREAD

**Issue by olartic** [itchio/itch.io#571]
When the app opens, I get a 403 error, and cannot see or do anything, which appears to be related to an Anti-DDoS service. If I open the link that comes up below in a browser:
[https://dl.itch.ovh/unarchiver/windows-386/LATEST?t=1480357883058]

what comes up is a Captcha page from Cloudflare, which, if I follow-through, gives me a download of the app. Something about DDoS-ing interference appears to be preventing site interoperability. In fact, this is also a problem on itch.io itself, but that is another matter. It would also be nice to be able to click that link and have it open a browser, because it's tedious to type it manually.

This completely prevents the app doing anything, regardless of various user efforts.

**Comment by leafo** [comment]
Our actual website doesn't have cloudflare in front of it, but the app downloads do. Can you show me an image of what you see when you get the 403 error?

**Comment by fasterthanlime** [comment]
Also helpful would be the country you're connecting to the internet from. Cloudflare has a bunch of blacklists in place by default.

**Comment by olartic** [comment]
US, via a VPN. Here are a few screenshots, one from the app, fresh after it is opened, the other two showing the itch homepage with broken images that, if you try to see their location, pull up the Captchas via Cloudflare. It does this regardless of browser, and I should add that I have attempted to close the VPN, and I still get this error.

I realize this is not the right forum for the main site troubles, but I have included a couple of screenshots there, because there is something going on, and you can plainly see it from my end, the images are all broken, for the entire site, but if I do that Captcha, they subsequently load (until they don't).

I appreciate you looking into this so quickly.

![403 error itch app](https://cloud.githubusercontent.com/assets/12633727/20686765/d2649036-b56e-11e6-900a-002b6768b9f2.png)
![itch io home page broken images](https://cloud.githubusercontent.com/assets/12633727/20686779/dc40c39a-b56e-11e6-9d7f-3358f3631c0b.png)
![itch io image captcha](https://cloud.githubusercontent.com/assets/12633727/20686786/e1da751c-b56e-11e6-895d-88e10a6b5de4.png)

**Comment by leafo** [comment]
Thanks for the image, as you noticed we do use cloudflare as a CDN for our images. I'm surprised they would block image URLs with a captcha, I'll see if I can disable it.

For app downloads, in the past we've whitelisted countries since many people were having regional issues. But sometimes it isn't a country issue, for example if CloudFlare's network thinks your IP address is associated with abuse then it may prompt a captcha. Is your VPN in a specific country, if so I can try white-listing it?

**Comment by olartic** [comment]
VPN is US also, typically. But I am confused as to why I can't bypass this in any way when using the app, to the extent that I cannot start anything, I cannot click something to open something, there is nothing from a UX perspective, that can be done to deal with this, other than bring this up on a GitHub comment page. I mean, I can complain to my VPN provider that they're enabling spammers, but that won't resolve this trouble.

Hopefully, there's a happy fix for everyone, here.

**Comment by Deozaan** [comment]
@fasterthanlime & @leafo Could you perhaps make the link clickable, and open in an itch tab, so that @olartic could do the captcha and have the cookie saved to the embedded itch browser? This should allow the itch app to work for olartic, at least temporarily.

**Comment by fasterthanlime** [comment]
@Deozaan apparently, it would still cut out randomly, see section of OP:

> but if I do that Captcha, they subsequently load (until they don't).

**Comment by Deozaan** [comment]
@fasterthanlime Right. I understood that. That's why I said it would work temporarily. I was under the impression that the images and things would work until the captcha cookie expired, and olartic would just have to enter the captcha again to get things working for a while longer.

**Comment by olartic** [comment]
It would be of some help, I think, because I would, at least, be able to update the app, or at least see what is going on! As it was, I had a meaningless error screen that I couldn't do anything about, as you can see on that screenshot. It is true, that images were not loading at all, but at least there was information to read and understand when I opened a browser window... text came up okay.

It is not much of an issue presently, since I've churned IPs, and complained to the VPN provider, but it does still come up periodically. However, the check completed at some point, so I can see what the expected behavior is to begin with. Additionally, as I am aware of this problem, I know, now, what causes it. Initially, it was very frustrating, since there was nothing I could do, and nothing intuitive to do, no help page came up, there was no alternate way to download, or check for a newer version, if there even was one! There is no way around that check, and since nothing loads, it's a black hole. Having a way to load the installed version page anyway might be a useful workaround, too--and, yeah, opening a browser for a Captcha would have been informative, if it didn't entirely solve things.

**Comment by fasterthanlime** [comment]
I agree with the above, but - the reason I'm hesitant to include workarounds is that I'd love to fix the actual cause of the issue (whitelisting countries, etc.) - although for VPNs, I don't know what''s doable.

If I were to add a workaround to the app, what would it look like? Detect 403 at setup time, and show a "Using a VPN?" paragraph with a link to open a window that gives you a chance to enter the CAPTCHA?
